compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -4152,7 +4246,46 @@ private void defineInvokableSymbolParams(BLangInvokableNode invokableNode, BInvo
             }
             paramSymbols.add(symbol);
         }
+        return paramSymbols;
+    }
 
+    public void defineInvokableTypeNode(BLangFunctionTypeNode functionTypeNode, long flags, SymbolEnv env) {
+        BInvokableTypeSymbol invokableTypeSymbol = (BInvokableTypeSymbol) functionTypeNode.symbol;
+        List<BVarSymbol> paramSymbols = defineParametersOfFunctionType(functionTypeNode, env);
+        invokableTypeSymbol.params = paramSymbols;
+
+        BType retType = null;
+        if (functionTypeNode.returnTypeNode != null) {
+            symResolver.resolveTypeNode(functionTypeNode.returnTypeNode, env);
+            invokableTypeSymbol.returnType = functionTypeNode.returnTypeNode.getBType();
+            retType = functionTypeNode.returnTypeNode.getBType();
+        }
+
+        BType restType = null;
+        if (functionTypeNode.restParam != null) {
+            defineNode(functionTypeNode.restParam, env);
+            invokableTypeSymbol.restParam = functionTypeNode.restParam.symbol;
+            restType = functionTypeNode.restParam.getBType();
+        }
+        List<BType> paramTypes = paramSymbols.stream()
```

Comment:
Not an issue, but can't we handle non-rest params, rest param, and then the return type. Just to maintain a natural order.